test/extensions/http/header_formatters/preserve_case/preserve_case_formatter_test.cc

```diff
@@ -9,7 +9,7 @@ namespace HeaderFormatters {
 namespace PreserveCase {
 
 TEST(PreserveCaseFormatterTest, All) {
-  PreserveCaseHeaderFormatter formatter;
+  PreserveCaseHeaderFormatter formatter(false);
```

Comment:
Can you verify that when this is false, and you set a reason phrase, you don't get a reason phrase back?

Reply:
Added another test case for this.